File: backend/app/model/manifest/MimeType.scala
package model.manifest

import org.neo4j.driver.v1.Value
import play.api.libs.json.{Format, Json}

case class MimeType(mimeType: String)

object MimeType {
  implicit val mimeTypeFormat: Format[MimeType] = Json.format[MimeType]

  def fromNeo4jValue(mimeType: Value): MimeType = {
    MimeType(
      mimeType = mimeType.get("mimeType").asString()
    )
  }
}

case class MimeTypeCoverage(mimeType: MimeType, humanReadableMimeType: Option[String],
                            total: Long, todo: Long, done: Long, failed: Long)

object MimeTypeCoverage {
  implicit val mimeCoverageTypeFormat: Format[MimeTypeCoverage] = Json.format[MimeTypeCoverage]
}

File: frontend/src/js/types/MimeType.ts
import PropTypes from "prop-types";

export const mimeTypePropType = PropTypes.shape({
  mimeType: PropTypes.string.isRequired,
});

export const mimeTypeCoveragePropType = PropTypes.shape({
  mimeType: mimeTypePropType.isRequired,
  humanReadableMimeType: PropTypes.string,
  total: PropTypes.number.isRequired,
  todo: PropTypes.number.isRequired,
  done: PropTypes.number.isRequired,
  failed: PropTypes.number.isRequired,
});

export interface MimeType {
  mimeType: string;
  priority: number;
}

export interface MimeTypeCoverage {
  mimeType: MimeType;
  humanReadableMimeType?: string;
  total: number;
  todo: number;
  done: number;
  failed: number;
}
